src/components/utils/src/logger/boostlogger.cc

```diff
@@ -0,0 +1,200 @@
+#include <boost/log/core.hpp>
+#include <boost/log/expressions.hpp>
+#include <boost/log/sinks/sync_frontend.hpp>
+#include <boost/log/sinks/text_file_backend.hpp>
+#include <boost/log/sinks/text_ostream_backend.hpp>
+#include <boost/log/sources/logger.hpp>
+#include <boost/log/sources/record_ostream.hpp>
+#include <boost/log/sources/severity_channel_logger.hpp>
+#include <boost/log/sources/severity_logger.hpp>
+#include <boost/log/support/date_time.hpp>
+#include <boost/log/trivial.hpp>
+#include <boost/log/utility/setup/common_attributes.hpp>
+#include <boost/log/utility/setup/file.hpp>
+#include <boost/log/utility/setup/from_settings.hpp>
+#include <boost/log/utility/setup/settings_parser.hpp>
+#include <fstream>
+
+#include <boost/date_time/c_local_time_adjustor.hpp>
+#include <boost/date_time/local_time_adjustor.hpp>
+
+#include <boost/regex.hpp>
+
+namespace logger {
+
+namespace logging = boost::log;
+namespace src = boost::log::sources;
+namespace sinks = boost::log::sinks;
+namespace keywords = boost::log::keywords;
+namespace expr = boost::log::expressions;
+namespace attrs = boost::log::attributes;
+
+BoostLogger::BoostLogger(const std::string& filename) : filename_(filename) {}
+
+void BoostLogger::Init() {
+  // Add formatting parameters to INI file
+  // Allows %Severity% to be used in ini config file for property Format.
+  boost::log::register_simple_formatter_factory<
+      boost::log::trivial::severity_level,
+      char>("Severity");
+  boost::log::register_simple_formatter_factory<std::string, char>("Component");
+  boost::log::register_simple_formatter_factory<boost::posix_time::ptime, char>(
+      "TimeStamp");
+  boost::log::register_simple_formatter_factory<std::thread::id, char>(
+      "ThreadId");
+  boost::log::register_simple_formatter_factory<std::string, char>("FileName");
+  boost::log::register_simple_formatter_factory<int, char>("LineNum");
+  boost::log::register_simple_formatter_factory<std::string, char>("Trace");
+
+  // Add filter parameters to INI file
+  // Allows %Severity% to be used in ini config file for property Filter.
+  boost::log::
+      register_simple_filter_factory<boost::log::trivial::severity_level, char>(
+          "Severity");
+  boost::log::register_simple_filter_factory<std::string, char>("Component");
+  boost::log::register_simple_filter_factory<std::string, char>("Trace");
+
+  std::ifstream file(filename_);
+  boost::log::settings settings = boost::log::parse_settings(file);
+
+  // Custom Settings
+
+  if (settings.has_section("Sinks")) {
+    auto& sinks = settings.property_tree().get_child("Sinks");
+    for (auto& sink : sinks) {
+      std::string sink_name = "Sinks." + sink.first;
+
+      // Disable logging for particular sinks
+      if (boost::optional<std::string> param =
+              settings[sink_name]["DisableLogging"]) {
+        bool sink_disabled = (param.get() == "true");
+        if (sink_disabled) {
+          sinks.erase(sink.first);
```

Comment:
calling erase invalidates the iterator we are looping on

Reply:
Fixed in https://github.com/smartdevicelink/sdl_core/pull/3571/commits/17a4d07d9c19313ad2bebda119e69f5825180956